src/python/grpcio/grpc/_server.py

```diff
@@ -769,10 +869,28 @@ class _ServerStage(enum.Enum):
 
 
 class _ServerState(object):
+    lock: threading.RLock
+    completion_queue: cygrpc.CompletionQueue
+    server: cygrpc.Server
+    generic_handlers: Optional[List[grpc.GenericRpcHandler]]
+    interceptor_pipeline: Optional[_interceptor._ServicePipeline]
+    thread_pool: futures.ThreadPoolExecutor
+    stage: _ServerStage
+    termination_event: threading.Event
+    shutdown_events: Sequence
```

Comment:
`List[threading.Event]`

Reply:
Is it possible to add other `Event` to this list? (Looks like we won't based on `_server.py`, but better to confirm).

Reply:
Yes. Only `_server.py` is allowed to access this API, so checking that `_server.py` is only using `threading.Event` is sufficient.